lib/ui/myplot/myplot_viewmodel.dart

```diff
@@ -26,7 +29,26 @@ class MyPlotViewModel {
         selectedCrop: store.state.myPlotState.selectedCrop,
         goToDetail: (CropEntity crop) => store.dispatch(GoToCropDetailAction(crop)),
         fetchCrops: () =>
-            store.dispatch(new FetchCropListAction())
+            store.dispatch(FetchCropListAction())
     );
   }
+
+  List<CropDetailProperty> getCropDetailProperties(CropEntity cropEntity) {
+    List<CropDetailProperty> listOfProperties = List();
+
+    if(cropEntity.complexity != null) {
+      String complexity = begAdvValues.reverse[cropEntity.complexity];
+      listOfProperties.add(CropDetailProperty(Strings.myPlotDetailComplexityTitle, Utils.capitalize(complexity)));
```

Comment:
It looks like the model of a property is only a single string and a list is being concatenated. If the widget display for list is to be changed to a different look fit for a list, then the model will not be able to cater to that change.

Should the CropDetailProperty be more flexible then?

Reply:
We should not over-engineer things that are not defined.
So far, we need to show properties of a crop...
If in the future we want to add more interaction, the change wont be difficult.
Right now, its better to keep it simple :)

Reply:
Also, to be added, the crop entity already holds all the data as enums.
This is a conversion to UI-ready component. 
So the full data its still at hand, if in the future we need an update.

Reply:
Fair point and I agree in keeping things simple. 

That said, I'm more worried about having a model that doesn't really represent the data structure. There's a possibility of other developers having to add processing to break up the data from the model. We then end up having:  
1. repository List data from firebase joined and converted into a single string stored in model
2. ui receives model data and then displays it, or split up the string into a list for a more specialized display.

This is likely when you have two different developers working on the UI and the repository.

Reply:
just saw additional comment. As long as the UI gets access to the data format in its original form then that's fine.

Reply:
Actually, by designing patterns of both android and ios, 
UI should never access the data format in its original form.
That's why we have "cropEntity" model and not simply "crop".

Its intended, that, at some point to have them differentiated.
An UI change will not always be a backend / data change and viceversa.

By design pattern, the UI should only now about the data it needs. 
The datamodel and the firebase manager usually has more information that UI doesnt need.
Timestamps, publication status and a few others are examples.